controllers/inferenceservice_controller.go

```diff
@@ -152,7 +150,24 @@ func (r *OpenshiftInferenceServiceReconciler) SetupWithManager(mgr ctrl.Manager)
 				}
 				return reconcileRequests
 			}))
-	err := builder.Complete(r)
+
+	// check if kserve is enabled, otherwise don't require Authorino.
+	enabled, err := utils.VerifyIfComponentIsEnabled(context.TODO(), r.client, utils.KserveAuthorinoComponent)
+	logMessage := "kserve service mesh component is enabled, Authorino is required"
+	if err != nil {
+		r.log.V(1).Error(err, "could not determine if kserve have service mesh enabled")
+		enabled = true
+		logMessage = "could not determine if kserve have service mesh enabled, Authorino integration will be enabled"
+	}
+
+	if enabled {
```

Comment:
Almost! More feedback:
The DSC can be:
```yaml
apiVersion: datasciencecluster.opendatahub.io/v1
kind: DataScienceCluster
metadata:
  name: default-dsc
spec:
  components:
    kserve: 
      managementState: "Removed",
      serving: 
        ingressGateway: 
          certificate: 
            type: SelfSigned
        name: "knative-serving",
        managementState: "Managed"
```
Notice kserve.managementState is `Removed` while serving.managementState is `Managed`. If KServe is removed, I think odh-model-controller don't need to care about authorino.

So, turns out we would need to check `kserve.managementState == Managed && serving.managementState != Removed`, to require authorino here.

I think I'm bad at explaining stuff.

Reply:
both enabled, right.

okay.